File: amenbank-frontend/src/app/models/scheduled-transfer.ts
export enum Frequency {
    DAILY = 'DAILY',
    WEEKLY = 'WEEKLY',
    MONTHLY = 'MONTHLY'
}

export interface ScheduledTransfer {
    id?: number;
    amount: number;
    startDate: string;
    endDate: string;
    frequency: Frequency;
    description: string;
    active: boolean;
    sourceAccountId: number;
    targetAccountNumber: string;
}

export interface ScheduledTransferRequest {
    amount: number;
    startDate: string;
    endDate: string;
    frequency: Frequency;
    description: string;
    sourceAccountId: number;
    targetAccountNumber: string;
}

export interface ScheduledTransferUpdateRequest {
    amount: number;
    startDate: string;
    endDate: string;
    frequency: Frequency;
    description: string;
    active: boolean;
    targetAccountNumber?: string;
} 

File: amenbank-backend/src/main/java/com/bankamen/entity/ScheduledTransfer.java
package com.bankamen.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class ScheduledTransfer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Double amount;

    private LocalDate startDate;
    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    private Frequency frequency; // DAILY, WEEKLY, MONTHLY

    private String description;

    private boolean active = true;

    @ManyToOne
    @JoinColumn(name = "source_account_id")
    private BankAccount sourceAccount;

    private String targetAccountNumber;

    // For backward compatibility
    @ManyToOne
    @JoinColumn(name = "target_account_id")
    private BankAccount targetAccount;
}
